src/Core/Checkout/Customer/Validation/Constraint/CustomerVatIdentificationValidator.php

```diff
@@ -54,23 +49,29 @@ public function validate(mixed $vatIds, Constraint $constraint): void
         }
     }
 
-    private function shouldCheckVatIdFormat(CustomerVatIdentification $constraint): bool
+    private function getVatIdPattern(CustomerVatIdentification $constraint): ?string
     {
-        if ($constraint->getShouldCheck()) {
-            return true;
-        }
-
-        return (bool) $this->connection->fetchOne(
-            'SELECT check_vat_id_pattern FROM `country` WHERE id = :id',
+        $vatIdInformation = $this->connection->fetchAssociative(
+            'SELECT check_vat_id_pattern, vat_id_pattern FROM `country` WHERE id = :id',
             ['id' => Uuid::fromHexToBytes($constraint->getCountryId())]
         );
-    }
 
-    private function getVatPattern(CustomerVatIdentification $constraint): string
-    {
-        return (string) $this->connection->fetchOne(
-            'SELECT vat_id_pattern FROM `country` WHERE id = :id',
-            ['id' => Uuid::fromHexToBytes($constraint->getCountryId())]
-        );
+        if ($vatIdInformation === false) {
+            return null;
+        }
+
+        \assert(\array_key_exists('check_vat_id_pattern', $vatIdInformation));
+        \assert(\array_key_exists('vat_id_pattern', $vatIdInformation));
```

Comment:
on most production systems, assertions are disabled. please make such checks with conditions, early returns or exceptions

Reply:
Mh, but those two array keys should always exist on a working environment? Otherwise the query should already fail?

Reply:
Sure, I'll be fine if you just remove them

Reply:
Then phpstan will complain :-D

Reply:
add a doctype above the query: `/** @var array{vat_id_pattern: string, check_vat_id_pattern: bool} $vatIdInformation */ :wink:

Reply:
As @mitelg suggested, please remove the asserts and use a null-safe operator instead